juju/bundle.py

```diff
@@ -627,7 +632,8 @@ async def run(self, context):
         # NB: this should really be handled by the controller when generating the
         # bundle change plan, and this short-term workaround may be missing some
         # aspects of the logic which the CLI client contains to handle edge cases.
-        if self.application in context.model.applications:
+        model = cast("Model", context.model)  # pyright: ignore[reportUnknownMemberType]
```

Comment:
This is just to take a step towards type checking for the `model._deploy` call site (and have my IDE be able to identify the call site!)

Reply:
It seems that the context argument is always an instance of BundleHandler, and typing the context and and a few fields in the BundleHandler would do just as well.

Then again, I'm wary of having any PR grow in scope until the entire library is typed, so I've created a separate ticket, #1214